manager/integration/tests/test_ha.py

```diff
@@ -2641,6 +2653,33 @@ def prepare_engine_not_fully_deployed_evnironment_with_volumes():
     5. Wait for the engine image CR state become deploying
     """
 
+    nodes = client.list_node()
+
+    volume1 = create_and_check_volume(client, "vol-1", size=str(3 * Gi))
+    volume2 = create_and_check_volume(client, "vol-2", size=str(3 * Gi))
+
+    core_api.patch_node(nodes[0].id, {
+        "spec": {
+            "taints":
+                [{"effect": "NoSchedule",
+                  "key": "key",
+                  "value": "value"}]
+        }
+    })
+
+    volume1.attach(hostId=nodes[0].id)
+    volume1 = common.wait_for_volume_healthy(client, volume1.name)
+    volume1.updateReplicaCount(replicaCount=2)
+
+    for r in volume1.replicas:
+        if r.hostId == nodes[0].id:
+            volume1.replicaRemove(name=r.name)
+            break
+
+    restart_and_wait_ready_engine_count(client, 2)
+
+    return volume1, volume2
```

Comment:
Before returning the volume1, volume2, can we first fetch the latest volume object by `client.by_id_volume(volume_name)`

Reply:
Hi @khushboo-rancher , updated!